File: app/src/main/java/com/example/mealresq/SplashScreen.java
package com.example.mealresq;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class SplashScreen extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        String checkbox = preferences.getString("remember", "");
        if (checkbox.equals("true")){
            Intent i = new Intent(SplashScreen.this, MainActivity.class);
            startActivity(i);
        } else{
            startActivity(new Intent(this, OnBoardActivity.class));
        }

    }
}

File: app/src/main/java/com/example/mealresq/SplashScreen.kt
package com.example.mealresq

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

class SplashScreen : AppCompatActivity() {
    private lateinit var auth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        auth = FirebaseAuth.getInstance()
    }
    override fun onStart() {
        super.onStart()
        if (auth.currentUser != null) {
            val intent = Intent(this@SplashScreen, MainActivity::class.java)
            startActivity(intent)
        } else{
            val intent = Intent(this@SplashScreen, OnBoardActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onBackPressed() {

    }
}